packages/ace-linters/language-provider.ts

```diff
@@ -331,7 +337,15 @@ class SessionLanguageProvider {
         if (!selectionRanges || selectionRanges[0].isEmpty()) {
             let row = this.session.getLength();
             let column = this.session.getLine(row).length - 1;
-            selectionRanges = [new AceRange(0, 0, row, column)];
+            selectionRanges =
+                [{
+                    start: {
+                        row: 0, column: 0
+                    },
+                    end: {
+                        row: row, column: column
+                    }
+                } as Ace.Range];
```

Comment:
why do you need to cast this as Ace.Range despite it not having any of the methods?

Reply:
We need this because of typescript type system - from `let selectionRanges = this.session.getSelection().getAllRanges();` we got `Ace.Range[]`, and without type-cast it would cause type-check errors, we could put `//@ts-ignore` before, but this is more preferable approach in that kind of situations

Reply:
I mean if we do not need the actual Range instance we can add new type definition AceRangeData and use it instead of trying to lie to the type system.
Or you could simply take range constructor from a range returned by getSelection something like `if (!AceRange) AceRange = editor.getSelectionRange().constructor`

Reply:
good point! I will make interface for places where we are not rely on instance